### PR TITLE
Iss344: A11y: Fix “Take Action” and “About” menu accessibility issues #344

### PR DESCRIPTION
**Pull request by Amberroseweeks**
### **A11y and other Updates to the Navigation Menu:**

Issue: https://github.com/CodeForPhilly/vacant-lots-proj/issues/344

Because of the site redesign some a11y issues no longer apply for desktop. I'd like to create a separate issue for mobile. For now I updated the desktop styling to match the new design. 

**Updated Colors**
- Added the gray-900 and updated gray-100 in the tailwind.config to match colors in Figma.

**IconLink Component:** 
- I changed this component to write a function to check if the NavbarItem isActive or not based on the url and pathname. If so isActive will be applied.
- I added aria-current to the Navbar item so it would read whether or not the user is on the current page depending on if the href matches the pathname. 
- If the NavbarItem isActive then the class active-state-nav will change the text color to green-700 matching the figma design. 
- I am also removing the aria label because link text is sufficient and you do not need an aria label for a link that already has link text. It's best to avoid unnecessary aria-labels. If these were standalone icons with no text then that would be needed.
- Removed the role of button for the links, buttons should be reserved for actions not navigations. Added the role of "link" because the component was not appearing in the link list, with the role it now appears in the ANDI link list view. This removes the spacebar functionality but links are expected to be activated with the ENTER key. Let me know if you want this added in anyway and we can write a function but I don't suggest marking the page navigation links as buttons. 

**Styling for Current Active Nav Item**
- Added two global styles. One to remove the white background to show the IsActive styles from the Navbar component. Another to hard code the active-state-nav link text color. I struggled to include this within the IsActive classes because the link color within the default IconLink component was overriding the green-700.

- For the header I included classes to style the isActive state for the Navbar, I am just importing IconLink since NavbarItem is wrapped around the IconLink component.

I'm new to NEXT UI so I struggled with the documentation a little but please let me know if there is anything that can be updated or refined here 

https://github.com/CodeForPhilly/vacant-lots-proj/assets/61482332/1487df09-8d1d-49e5-9dfb-9b4b17f4ff8d

https://github.com/CodeForPhilly/vacant-lots-proj/assets/61482332/fee4a276-8e7b-4867-959c-4671165b690c

### CODE REVIEW

**Review by marvieqa — APPROVED** [review]
Looks good @Amberroseweeks ! The top navigation links are announcing beautifully using a screen reader! Good to ship!

**Review by brandonfcohen1 — CHANGES_REQUESTED** [review]
code looks good, but looks like linting isn't properly set up. check out here: https://github.com/CodeForPhilly/vacant-lots-proj/blob/main/docs/SETUP/FRONT_END.md#formatting

**Review by bacitracin — APPROVED** [review]
Very nice!

**Review by markgoetz — COMMENTED** [review]
Had a couple of software engineering thoughts.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "editor.formatOnSave": true,
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
+}
```
[the remark this answers is not in the document]
Comment (Amberroseweeks) [reply .vscode/settings.json:4]: Thank you for this great feedback @markgoetz ! I am just unsure of how to implement this, I followed the front end instructions but i'm not sure what new line to add to this file. I will commit the other changes for now though.

[the remark this answers is not in the document]
Comment (Amberroseweeks) [reply .vscode/settings.json]: Thanks @BayLadyCoder for helping me figure this one out, I added another line and just pushed the changes

**Comment by nlebovits** [comment]
@brandonfcohen1 ready for merge?

**Review by brandonfcohen1 — APPROVED** [review]

**Pull request merged**